python/pyspark/sql/profiler.py

```diff
@@ -158,6 +159,70 @@ def _profile_results(self) -> "ProfileResults":
         """
         ...
 
+    def dump_perf_profiles(self, path: str, id: Optional[int] = None) -> None:
```

Comment:
I wonder if we should have something like `spark.profile.memory` or `spark.profile.dump(type="memory")`, `spark.profile.show(type="memory")`

Reply:
Good point!
V1 has `sc.dump_profiles/show_profiles` for both perf and memory profiling.
V2 has `spark.dumpPerfProfiles` and `spark.dumpMemoryProfiles` for perf and memory profiling separately.
It would be more consistent and user-friendly to introduce a uniform interface for both like `spark.profile.dump/show`.
Let me create a ticket for now.
What's your thought on that @ueshin?

Reply:
Sounds good.